variety/plugins/builtin/downloaders/WallhavenDownloader.py

```diff
@@ -138,13 +143,7 @@ def fill_queue(self):
                 # missing or unparseable resolution - consider ok
                 pass
 
-            try:
-                link = thumb.find("a", {"class": "preview"})["href"]
-                if self.is_in_banned(link):
```

Comment:
This is a check that is lacking in the new code that we definitely need - it prevents downloading again images that the user has manually moved to Trash. The banned list contains "origin_url"-s, i.e. the URL of the page that hosts the image, not the URL of the image itself.

Reply:
Yes, I should not delete those lines.